### PR TITLE
Enable running individual passes in the driver

### PR DESCRIPTION
**Pull request by pfultz2**

### CODE REVIEW

File: src/driver/passes.cpp [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+#include <migraphx/simplify_reshapes.hpp>
+
+#include <migraphx/ranges.hpp>
+#include <unordered_map>
+
+namespace migraphx {
+namespace driver {
+inline namespace MIGRAPHX_INLINE_NS {
+
+std::unordered_map<std::string, pass> create_passes_lookup()
```
Comment (umangyadav) [inline src/driver/passes.cpp:58]: Enumerating passes would add more maintenance. Can we create Registry of available passes ?

File: src/driver/main.cpp [hunk 2]
```diff
@@ -123,6 +125,7 @@ struct loader
            ap.append(),
            ap.nargs(2));
         ap(optimize, {"--optimize", "-O"}, ap.help("Optimize when reading"), ap.set_value(true));
+        ap(passes, {"--apply-pass", "-p"}, ap.help("Passes to apply to model"), ap.append());
```
Comment (umangyadav) [inline src/driver/main.cpp:128]: Need to add this option in documents

File: src/driver/main.cpp [hunk 3]
```diff
@@ -330,6 +333,8 @@ struct loader
                                      migraphx::dead_code_elimination{},
                                  });
         }
+        if(not passes.empty())
```
Comment (umangyadav) [inline src/driver/main.cpp:336]: Side note : `optimize` option can utilize `optimize_module` passs.

File: src/driver/passes.cpp (hunk 1 above)
Comment (pfultz2) [reply src/driver/passes.cpp:58]: We can try to do that in the future, but we would need to make the passes serializable, but I think thats beyond the scope of this PR.

File: src/driver/main.cpp (hunk 2 above)
Comment (pfultz2) [reply src/driver/main.cpp:128]: Updated.

**Review by umangyadav — APPROVED** [review]

**Pull request merged**